feat(Rows): simplify component's logic, implement withChevron prop and allow right with controls

marcoskolodny opened the pull request:
Issues:
- [WEB-1549](https://jira.tid.es/browse/WEB-1549)
- [WEB-1777](https://jira.tid.es/browse/WEB-1777)
- [WEB-1857](https://jira.tid.es/browse/WEB-1857)

To do list:
- [X] Reimplement row's logic
- [X] Allow right + controls
- [X] Fix logic for withChevron prop
- [X] Make usage of role/dataAttributes/ref consistent between different use cases
- Verify that projects using Rows are not affected by these changes:
    - [X] webapp: all tests passed (https://github.com/Telefonica/webapp/actions/runs/9035580127)
    - [X] CMS: the Mistica's version in the repo is too old, and when I try to upgrade it there are several things that fail. I've notified the CMS team about it so that they can take a look into it.
    - [X] global checkout: tested locally by replacing `node_modules` in `global-checkout-webview` repo. There were no breaking changes. There are several screenshot tests failing, but it's because that repo has an outdated version of Mística, and it doesn't contain latest changes related to rows. I've created two PRs that will fix the issues related to this:
        - https://github.com/Telefonica/global-checkout-webview/pull/420
        - https://github.com/Telefonica/global-checkout/pull/1012).

marcoskolodny commented on src/icon-button.tsx:
```diff
@@ -127,6 +127,7 @@ interface BaseProps {
     bleedLeft?: boolean;
     bleedRight?: boolean;
     bleedY?: boolean;
+    role?: string;
```
Unrelated. Added role prop to iconButton (other buttons have it)

marcoskolodny commented on src/list.tsx:
```diff
@@ -53,27 +55,26 @@ interface CommonProps {
 
     return centerY ? (
-        <div style={{display: 'flex', alignItems: 'center', height: '100%'}}>{right}</div>
+        <div style={{display: 'flex', flexDirection: 'column', justifyContent: 'center', height: '100%'}}>
```
This change fixes [this bug](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AlCB3AOgOwAS4BcBLAgGxgF5MQAVE86nfAJyIHMALAi4J-XJAD4%2B-AVCIA3QQRgAPArgCMSAPTipI-knXS5CgEyqdm1cLy4AvjhVmcqDH2JlK1Os8bnYAZzCsADsQQ2FQgACIwPv6B2B4s7Fw8Jmai%2BNqSuvJKRumaqToymYZqOeapNnxW2OXYIAA0IAQcMAC2EQgA2iAAshASRF4EAIbMIAC69ehEUI1eHQDM%2BgAMoxZAA)

marcoskolodny commented on src/list.tsx:
```diff
@@ -386,22 +367,33 @@ const RowContent = React.forwardRef<TouchableElement, RowContentProps>((props, r
+        onNavigate: props.onNavigate,
+        onPress: props.onPress,
+        trackingEvent: props.trackingEvent,
+    } as unknown as TouchableProps;
```
If there is a cleaner way to do this, I'm all ears :)

atabel replied:
the alternative way is as it is in master, not merging the to/href/onPress cases. Another option that I think could work is by destructuring all props but interactiveProps, something like this: https://github.com/Telefonica/mistica-web/blob/master/src/text-link.tsx#L40

marcoskolodny replied:
Discussed offline. I've removed `as unknown` to avoid weakening TS type safety.

marcoskolodny commented on src/list.tsx:
```diff
@@ -53,27 +55,26 @@ interface CommonProps {
     return centerY ? (
-        <div style={{display: 'flex', alignItems: 'center', height: '100%'}}>{right}</div>
+        <div style={{display: 'flex', alignItems: 'center', height: '100%'}}>
+            <div>{right}</div>
```
wrapping this inside a div, because of [this case](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AlCB3AOgOwAS4BcBLAgGxgF5MQAVE86nfCbABQCcYBnLi4ACgCUuCgD5cwAL6SmudkQDmACwJ9Z%2BJKPX5cSKEQBuogtwIBGJAHp9R7RpvHTAJisPtVrXlwzslzyAAaEAIlGABbbgQAbRAAWQgDIi4CAEN2EABdIPQiKBCuaIBmJwAGDMkgA). We don't want to force `display: flex` in the extra content by default. By wrapping `right` inside an interior div, we avoid this (but we center the content).

yceballost reviewed (approved):
I don't really like the idea to have the possibility of remove chevron icon with href, but..

atabel reviewed (commented):
I miss some new cases in `list-type-test.tsx` covering the changes in this PR

atabel commented:
@Marcosld, this should unlock this kind of UX (https://jira.tid.es/browse/WEB-1549):
https://mistica-fkrtebrsi-flows-projects-65bb050e.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AhCAPGUBKEDuAOgHYAEJALgJbkA2MAvASACIQkDOEAtjOQBaUiAcybEyEIgAUATjHbt6wABQBKEvQB8JYGLIdeAZXIBDcjCVMaEY1EEiQAGgrSArjBUBuXWXa8AKpQ8EC7kSqrqWjqkej6GJmYWIFY2dkxOAGbGNL6e3npZMNKhwCQ88sZCMPAkTKxEMACETCQAvrnRZC1OAEwADP3tnS26tuzGAEZ0UIqV5Eam5pbWtsJMKsPReNR8AMJ8MABu0hKKDbPzCUspqyDrutKUQnzkM3ELickr9moA-CRIBgADoJ6tISLAaMYAJ6MEC9LjsZoAei01SILhoNA2KOIjhA-BgZQQAG0QAB5bogAC6DhAWyg-HYJIAzAA2XpUlpAA

marcoskolodny commented:
> I miss some new cases in `list-type-test.tsx` covering the changes in this PR

I've added some tests covering scenario of right+controls. I've also added some tests related to withChevron prop, but this is not something introduced in this PR. This prop already existed, but it had no effect in the component

pladaria commented:
> **Size stats**
> 
> master	this branch	diff
> Total JS	10.7 MB	10.7 MB	-3.3 kB
> JS without icons	1.93 MB	1.92 MB	-3.3 kB
> Lib overhead	52.1 kB	52.1 kB	0 B
> Lib overhead (gzip)	14 kB	14 kB	0 B

thats nice!

tuentisre commented:
:tada: This PR is included in version 15.9.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@telefonica/mistica/v/15.9.0)
- [GitHub release](https://github.com/Telefonica/mistica-web/releases/tag/v15.9.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
